opm/simulators/wells/SingleWellState.cpp

```diff
@@ -190,11 +182,29 @@ void SingleWellState::update_producer_targets(const Well& ecl_well, const Summar
         assert(this->pu.phase_used[BlackoilPhases::Vapour]);
         this->surface_rates[pu.phase_pos[BlackoilPhases::Vapour]] = -prod_controls.gas_rate;
         break;
+    case Well::ProducerCMode::GRUP:
+    case Well::ProducerCMode::THP:
+    case Well::ProducerCMode::BHP:
+        if (this->pu.phase_used[BlackoilPhases::Liquid]) {
+            this->surface_rates[pu.phase_pos[BlackoilPhases::Liquid]] = -1000.0 * Opm::unit::cubic(Opm::unit::meter) / Opm::unit::day;
+        }
+        if (this->pu.phase_used[BlackoilPhases::Aqua]) {
+            this->surface_rates[pu.phase_pos[BlackoilPhases::Aqua]] = -1000.0 * Opm::unit::cubic(Opm::unit::meter) / Opm::unit::day;
+        }
+        if (this->pu.phase_used[BlackoilPhases::Vapour]){
+            this->surface_rates[pu.phase_pos[BlackoilPhases::Vapour]] = -100000.0 * Opm::unit::cubic(Opm::unit::meter) / Opm::unit::day;
+        }
+        break;
+
```

Comment:
It is difficult to evaluate when the hard-coded values will work or will not work. But I guess compared with the original master branch code, which these surface rates zeros, I do not think these hard-coded code will do worse.